### PR TITLE
Allow migrations to produce additional commands

### PR DESCRIPTION
**Pull request by smerrell**
Alter.Collection can be used now to add additional ICommandData to the
Alter.Collection batch. This makes it so you can safely add new
documents or make changes to additional documents and have it be part of
the same batch / transaction.

This changes the API of Alter.Collection though, so this might need to
be changed to allow for both the Action and the Func.

### CODE REVIEW

**Comment by smerrell** [comment]
This could use some discussion since it makes a pretty big change to Alter.Collection.

**Comment by dportzline83** [comment]
This change allows "additional" changes that need to happen along with the changes to the collection to be batched in the *same* transaction. This is very important. You don't want changes that rely on each other to pass/fail independently. :+1:

**Comment by smerrell** [comment]
Indeed. I'm curious to see what people think about the API changes. We've gone from an `Action<RavenJObject, RavenJObject>` to a `Func<RavenJObject, RavenJObject, IEnumerable<ICommandData>>`. This is a pretty big API change so I'm curious if others would want to have the old API around as well or if it is ok to just change from the Action to the Func.

**Comment by dportzline83** [comment]
Yep, this would be a breaking change and would require a major version bump. Is there a better way to do this to make it backwards compatible? Or is this the cleanest way?

File: RavenMigrations.Tests/AlterTests.cs [hunk 1]
```diff
@@ -57,6 +59,30 @@ public void Can_migrate_up_to_new_clr_type()
...
+            {
+                InitialiseWithPerson(store, "Sean Kearon");
+
+                var migration = new AlterCollectionMigration();
+                migration.Setup(store);
+
+                migration.Up();
+                WaitForIndexing(store);
+
+                Thread.SpinWait(100000000);
```
Comment (dportzline83) [inline RavenMigrations.Tests/AlterTests.cs]: I don't think this needs to be here. Not sure why it was in the other test. The ```WaitForIndexing(store)``` call on the previous line might not be needed either.

**Comment by dportzline83** [comment]
Merged as part of #33.

**Pull request closed without merging**